include/circt/Dialect/Ibis/IbisOps.td

```diff
@@ -160,13 +167,48 @@ def CallOp : IbisOp<"call", [CallOpInterface]> {
   }];
 }
 
+def GetParentOp : IbisOp<"get_parent"> {
+  let summary = "Ibis get parent";
+  let description = [{
+    Given an Ibis class reference, returns the parent of said class. The
```

Comment:
Should the description say what happens if there is no parent?

Reply:
I can add it - as with most things parent related it'll be "won't be detectable until some pass which looks more closely at the instance hierarchy resolves parents". There might also be merit in having an explicit instance hierarchy verification pass which opt-in validates this (i.e. not part of the op verifiers).